### DISCUSSION TITLE
Error on gnupghome should be a directory

### DISCUSSION THREAD

**Discussion by GenesisAdam** [simple-login/app#1219] [Self-hosting question]
I am trying to install this on a synology nas as slef-hosted.
I am using the simplelogin/app:4.6.2-beta.
When I run the sl-init docker, I get the following error:

Traceback (most recent call last):
  File "init_app.py", line 8, in <module>
    from app.pgp_utils import load_public_key
  File "/code/app/pgp_utils.py", line 14, in <module>
    gpg = gnupg.GPG(gnupghome=GNUPGHOME)
  File "/usr/local/lib/python3.7/site-packages/gnupg.py", line 827, in __init__
    raise ValueError('gnupghome should be a directory (it isn\'t): %s' % gnupghome)
ValueError: gnupghome should be a directory (it isn't): /volume1/docker/simplelogin/gpg

Checking my gpg home by running gpg --version, I can confirm that the gnupg home is correctly set as seen below:

gpg (GnuPG) 2.2.18
libgcrypt 1.9.4
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /volume1/docker/simplelogin/gpg

Any help will be greatly appreciated.

**Comment by nguyenkims** [comment]
Do you explicitly set `GNUPGHOME` variable in your config file? By default, it uses a random directory in `/tmp`. The `gpg` used by SimpleLogin doesn't have the same config as the `gpg` cli tool.

**Reply by GenesisAdam** [reply]
Yes, it was set in the simplelogin.env file and that is where it's picking up the value but it just won't recognise it as a valid directory.
Or is there another config file where the GNUPGHOME variable is required to be set for SL?
I have repeated this install 5 times since I raise this issue and is still the same error message, I am about to give up.

**Reply by nguyenkims** [reply]
Do you have the docker volume mounted correctly? By default SimpleLogin uses `/sl/pgp` to store PGP keys and this directory is mounted via `-v $(pwd)/sl:/sl \` parameter in Docker.

**Reply by GenesisAdam** [reply]
Thanks for your help. I have sorted out the directory mapping and the error is no longer occuring.

**Comment by GenesisAdam** [comment]
All volumes are mapped and if not docker will flag error. What is the latest stable version that I can install.
When Itried using the "latest", I was getting error. For example this code:
`sudo docker run --rm \
    --name sl-init \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/upload:/code/static/upload \
    -v $(pwd)/simplelogin.env:/code/.env \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    --network="sl-network" \
    simplelogin/app:latest python init_app.py`

Returned an error:

python: can't open file 'init_app.py': [Errno 2] No such file or directory
